pkg/tenant/db/tenants.go

```diff
@@ -30,18 +31,16 @@ func (t *Tenant) Namespace() string {
 }
 
 func (t *Tenant) MarshalValue() ([]byte, error) {
-	// TODO: look into bson, msgpack, etc.
-	return json.Marshal(t)
+	return msgpack.Marshal(t)
 }
 
 func (t *Tenant) UnmarshalValue(data []byte) error {
-	// TODO: look into bson, msgpack, etc.
-	return json.Unmarshal(data, t)
+	return msgpack.Unmarshal(data, t)
 }
 
 func CreateTenant(ctx context.Context, tenant *Tenant) (err error) {
-	if tenant.ID == uuid.Nil {
-		tenant.ID = uuid.New()
+	if tenant.ID.String() == "" {
+		tenant.ID = ulid.Make()
```

Comment:
I used Make() for now, but am thinking I should probably replace with New() as mentioned in the ReadMe here: https://github.com/oklog/ulid

What do you think?

Reply:
Yes, you're probably right -- let's create a story for this. I think we could create a small `ulid` package that wraps the `oklog/ulid` package and has the `IsZero()` functionality mentioned above and also has a `Make` function that uses cryptographic security and monotonic entropy. We'll also probably have to ensure thread safety.